Wikipedia/UI-V5/WMFSaveButtonController.m

```diff
@@ -0,0 +1,83 @@
+//  Created by Monte Hurd on 8/19/15.2//  Copyright (c) 2015 Wikimedia Foundation. Provided under MIT-style license; please copy and modify!
+
+#import "WMFSaveButtonController.h"
+#import "MWKSavedPageList.h"
+#import "MWKArticle.h"
+#import "SessionSingleton.h"
+#import "MWKUserDataStore.h"
+
+@interface WMFSaveButtonController ()
+
+@property (strong, nonatomic) UIButton* button;
+
+@end
+
+@implementation WMFSaveButtonController
+
+- (instancetype)initWithButton:(UIButton*)button
+                         title:(MWKTitle*)title {
+    self = [super init];
+    if (self) {
+        self.button = button;
+        self.title  = title;
+        [self observeSavedPages];
+        [self updateSavedButtonState];
+    }
+    return self;
+}
+
+- (MWKSavedPageList*)savedPageList {
+    return [SessionSingleton sharedInstance].userDataStore.savedPageList;
```

Comment:
@montehurd Couple of notes…

1. Use injection whenever possible. Accessing singletons should be a pattern of last resort. If you have to access the `SessionSingleton` in a class… consider that a code smell. Try to pass anything required by a class on initialization. In this case, you should be passing the required`MWKSavedPageList` in the initializer. 

2. Re: "Central Saved page List" - currently we are using only one saved page list in the app. The one on the session singleton. This should be the one that is passed in. So you don't need to worry about observing the "correct one" - just observe the one that is passed in.